src/Chapter3.hs

```diff
@@ -971,6 +1087,29 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek 
+    = Sun
+    | Mon 
+    | Tue 
+    | Wed 
+    | Thu 
+    | Fri 
+    | Sat deriving (Show, Read, Eq, Ord, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Sat = True
+isWeekend Sun = True
+isWeekend _ = False 
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sat = Sun
```

Comment:
That is right!
You could also use the instance of `Bounded` here:

```suggestion
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂